src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -372,13 +376,13 @@ public async ValueTask<int> ReadDataAsync(Memory<byte> buffer, CancellationToken
                     return 0;
                 }
 
-                (Task waitForData, int bytesRead) = TryReadFromBuffer(buffer.Span);
-                if (waitForData != null)
+                (bool wait, int bytesRead) = TryReadFromBuffer(buffer.Span);
+                if (wait)
                 {
                     Debug.Assert(bytesRead == 0);
-                    await waitForData.ConfigureAwait(false);
-                    (waitForData, bytesRead) = TryReadFromBuffer(buffer.Span);
-                    Debug.Assert(waitForData == null);
+                    await AsConfiguredAwaitable();
```

Comment:
This is super picky, but I suspect it will trip me up to see this "await" without ConfigureAwait(false) after it.

Reply:
Ok, I'll remove it from GetWaiterTask and just let the call site do it.